aspnetcore/fundamentals/hosted-services/sample/Services/TimedHostedService.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace BackgroundTasksSample.Services
+{
+    #region snippet1
+    internal class TimedHostedService : IHostedService, IDisposable
+    {
+        private Timer _timer;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is starting.");
+
+            _timer = new Timer(DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is working.");
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is stopping.");
+
+            _timer.Change(Timeout.Infinite, 0);
```

Comment:
You need to handle the timer being null here

Reply:
Ok, but that's not what the doc I read said to do. They say to do it using `Change`. Do you want to see the reference?

Reply:
> If dueTime is negative one (-1) milliseconds, the callback method is never invoked; the timer is disabled.

I think u know! lol `Timeout.Infinite` = -1

> When a timer is no longer needed, use the Dispose method to free the resources held by the timer.

Reply:
@davidfowl ? Stop shouldn't be called without Start.

Reply:
This is true. It's only if StartAsync fails before the assignment (which can't happen in this case) that we'll still call StopAsync and it would be null.

Reply:
sry ... I understand now.